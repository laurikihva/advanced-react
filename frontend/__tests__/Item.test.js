import { shallow } from 'enzyme';

import ItemComponent from '../components/Item';

const fakeItem = {
    id: 'ABC123',
    title: 'Awesome Test Item',
    price: 5000,
    description: 'This cool item is made for testing!',
    image: 'dog.jpg',
    largeImage: 'largedog.jpg',
};

describe('<Item/>', () => {
    it('renders the pricetag and title properly', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem} />);
        const priceTag = wrapper.find('PriceTag');
        const title = wrapper.find('Title a');

        expect(priceTag.children().text()).toBe('â‚¬50');
        expect(title.text()).toBe(fakeItem.title);
    });
    
    it('renders the image properly', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem} />);
        const img = wrapper.find('img');
        
        expect(img.props().src).toBe(fakeItem.image);
        expect(img.props().alt).toBe(fakeItem.title);
    });

    it('renders out the buttons properly', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem} />);
        const buttonList = wrapper.find('.buttonList');

        expect(buttonList.children()).toHaveLength(3);
        expect(buttonList.find('Link')).toHaveLength(1);
        expect(buttonList.find('AddToCart').exists()).toBe(true);
        expect(buttonList.find('DeleteItem').exists()).toBe(true);
    })
});
