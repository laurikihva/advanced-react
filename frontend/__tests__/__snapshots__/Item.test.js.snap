// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Awesome Test Item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Awesome Test Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    â‚¬40
  </PriceTag>
  <p>
    This cool item is made for testing!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete this Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
